platform/fabric/core/generic/committer/endorsertx.go

```
@@ -18,83 +18,107 @@ import (
+	switch vc {
+	case driver.Valid:
+		// TODO: this might be due the fact that there are transactions with the same tx-id, the first is valid, the others are all invalid
+		logger.Warnf("transaction [%s] in block [%d] is marked as valid but for fabric is invalid", txID, blockNum)
+	case driver.Invalid:
+		if logger.IsEnabledFor(zapcore.DebugLevel) {
+			logger.Debugf("transaction [%s] in block [%d] is marked as invalid, skipping", txID, blockNum)
 		}
-	default:
+		// Nothing to commit
+	case driver.Unknown:
 		if logger.IsEnabledFor(zapcore.DebugLevel) {
-			logger.Debugf("transaction [%s] in block [%d] is not valid for fabric [%s], discard!", txID, blockNum, validationCode)
+			logger.Debugf("transaction [%s] in block [%d] is marked as unknown, skipping", txID, blockNum)
 		}
-
-		vc, deps, err := committer.Status(txID)
+		// Nothing to commit
+	default:
+		event.Err = errors.Errorf("transaction [%s] status is not valid: %d", txID, validationCode)
+		err = committer.DiscardTx(event.Txid)
 		if err != nil {
-			logger.Panicf("failed getting tx's status [%s], with err [%s]", txID, err)
-		}
-		event.DependantTxIDs = append(event.DependantTxIDs, deps...)
-		switch vc {
-		case driver.Valid:
-			// TODO: this might be due the fact that there are transactions with the same tx-id, the first is valid, the others are all invalid
-			logger.Warnf("transaction [%s] in block [%d] is marked as valid but for fabric is invalid", txID, blockNum)
-		case driver.Invalid:
-			if logger.IsEnabledFor(zapcore.DebugLevel) {
-				logger.Debugf("transaction [%s] in block [%d] is marked as invalid, skipping", txID, blockNum)
-			}
-			// Nothing to commit
-			return
-		default:
-			event.Err = errors.Errorf("transaction [%s] status is not valid: %d", txID, validationCode)
-			err = committer.DiscardTx(event.Txid)
-			if err != nil {
-				logger.Errorf("failed discarding tx in state db with err [%s]", err)
-			}
+			logger.Errorf("failed discarding tx in state db with err [%s]", err)
```

are we ignoring this error on purpose? if that is the case should we change to `Debugf` or `Warnf`?